### PR TITLE
Search quiz by name

### PR DESCRIPTION
**Pull request by andreiuta97**
Make search quiz by name with abstract methods.
Not all text has to be written to have results.

### CODE REVIEW

File: src/Service/QuizTemplateService.php [hunk 1]
```diff
@@ -46,15 +46,16 @@ public function getQuiz(int $id): QuizTemplate
         return $quiz;
     }
```
Comment (victortodoran) [inline src/Service/QuizTemplateService.php:48]: I would rename this method to be more transparent with what it does. getFilteredQuizez or getQuizezByFilter, getQuizezFilterdQuizezForPage etc. getQuizzes leads me to believe all quizez are returned

File: src/Service/QuizTemplateService.php [hunk 2]
```diff
@@ -46,15 +46,16 @@ public function getQuiz(int $id): QuizTemplate
...
 
-    public function getQuizzes(int $currentPage): array
+    public function getQuizzes(array $filters, int $currentPage): array
     {
-        $criteria = new Criteria([], [], ($currentPage - 1) * 5, 5);
-        return $this->quizTemplateRepo->findBy($criteria);
+        $criteria = new Criteria($filters, [], ($currentPage - 1) * 5, 5);
+        return $this->quizTemplateRepo->findBySearch($criteria);
     }
```
Comment (victortodoran) [inline src/Service/QuizTemplateService.php:56]: getFilteredQuizezNumber maybe?

File: src/Service/QuizTemplateService.php [hunk 3]
```diff
@@ -46,15 +46,16 @@ public function getQuiz(int $id): QuizTemplate
         return $quiz;
     }
 
-    public function getQuizzes(int $currentPage): array
+    public function getQuizzes(array $filters, int $currentPage): array
     {
-        $criteria = new Criteria([], [], ($currentPage - 1) * 5, 5);
-        return $this->quizTemplateRepo->findBy($criteria);
+        $criteria = new Criteria($filters, [], ($currentPage - 1) * 5, 5);
```
Comment (calinpristavu) [inline src/Service/QuizTemplateService.php]: 6

**Review by calinpristavu — CHANGES_REQUESTED** [review]

File: src/Repository/QuizTemplateRepository.php [hunk 4]
```diff
@@ -3,39 +3,24 @@
...
-    public function getNumberOfQuizzes(): int
-    {
-        $sql = 'SELECT count(*) as quizzesNumber FROM quiz_template';
-        $dbStmt = $this->pdo->prepare($sql);
-        $dbStmt->execute();
-
-        return $dbStmt->fetch(\PDO::FETCH_COLUMN);
-    }
-
     public function saveQuestionsForQuiz($id, $questions)
```
Comment (victortodoran) [inline src/Repository/QuizTemplateRepository.php:11]: Let us have some consistency for variable names please. Choose one name for statements ($dbStmt or $sqlStm) and use it throughout. 
In a future iteration when making multiple calls to the database which are co-dependent please consider using transactions. If you received smth better from trainers. Consider using that.
https://www.php.net/manual/en/pdo.begintransaction.php
https://www.php.net/manual/en/pdo.commit.php
https://www.php.net/manual/en/pdo.rollback.php

**Review by calinpristavu — APPROVED** [review]

**Pull request merged**